cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -286,10 +321,78 @@ public ConfigMap generateMetricsConfigMap() {
         if (isMetricsEnabled) {
             volumeMountList.add(createVolumeMount(metricsConfigVolumeName, metricsConfigMountPath));
         }
+        if (rackConfig != null) {
+            volumeMountList.add(createVolumeMount(KAFKA_INIT_VOLUME_NAME, KAFKA_INIT_VOLUME_MOUNT));
+        }
 
         return volumeMountList;
     }
 
+    @Override
+    protected Affinity getAffinity() {
+
+        List<WeightedPodAffinityTerm> weightedPodAffinityTerms = new ArrayList<>();
+        PodAntiAffinity podAntiAffinity = null;
+        Affinity affinity = null;
+
+        // adding the affinity term for rack feature only if it's enabled
+        if (rackConfig != null) {
+            Map<String, String> matchLabels = new HashMap<>();
+            matchLabels.put(Labels.STRIMZI_CLUSTER_LABEL, cluster);
+            matchLabels.put(Labels.STRIMZI_NAME_LABEL, name);
+            matchLabels.put(Labels.STRIMZI_TYPE_LABEL, AssemblyType.KAFKA.toString());
+            LabelSelector labelSelector = new LabelSelector(null, matchLabels);
+
+            PodAffinityTerm podAffinityTerm = new PodAffinityTerm(labelSelector, null, rackConfig.getTopologyKey());
+            WeightedPodAffinityTerm weightedPodAffinityTerm = new WeightedPodAffinityTerm(podAffinityTerm, 100);
+            weightedPodAffinityTerms.add(weightedPodAffinityTerm);
+        }
+
+        if (weightedPodAffinityTerms.size() > 0) {
+            podAntiAffinity = new PodAntiAffinityBuilder()
+                    .withPreferredDuringSchedulingIgnoredDuringExecution(weightedPodAffinityTerms)
+                    .build();
+        }
+
+        if (podAntiAffinity != null) {
+            affinity = new AffinityBuilder()
+                    .withPodAntiAffinity(podAntiAffinity)
+                    .build();
+        }
+
+        return affinity;
+    }
+
+    @Override
+    protected List<Container> getInitContainers() {
+
+        List<Container> initContainers = new ArrayList<>();
+
+        if (rackConfig != null) {
+
+            List<EnvVar> varList = new ArrayList<>();
```

Comment:
`Arrays.asList()`

Reply:
If we are moving on `Arrays.asList()` I'll do the same in all the other parts where we are using `new ArrayList<>()` (i.e. `getVolumes`, `getVolumeClaims`, ...) is that ok?

Reply:
It's just much easier to read in a case like this were we know there will be a fixed number of elements in the list. One expression, rather than needing several statements.

Reply:
Sorry I misleading what you meant. Totally agree. Done ;)